src/streaming/models/FragmentModel.js

```diff
@@ -227,6 +235,28 @@ MediaPlayer.dependencies.FragmentModel = function () {
                     return isNaN(req1.index) && isNaN(req2.index) && (req1.quality === req2.quality);
                 },
 
+                isDiscarded = function() {
+                    var buffer = this.videoModel.getElement(),
+                        inBuffer = this.sourceBufferExt.getBufferRange(buffer, request.availabilityStartTime) !== null,
+                        req,
+                        d;
+
+                    // It can take a few moments to get into the buffer
+                    if (!inBuffer) {
+                        d = new Date();
+                        d.setSeconds(d.getSeconds() + 3);
```

Comment:
Can you please explain this 3 seconds. Is it related to the chunk duration? Or is it for edge cases?